Refactor Browse string to Choose File

roidayan opened the pull request:
Browse is not part of the Wordpress translatation file so it doesn't get
translated. Choose File is and will be translated according to the
langauge Wordpress is set.

Signed-off-by: Roi Dayan <roi.dayan@gmail.com>

roidayan commented:
The suggestion here is because Browse is being translated with the default Wordpress text domain so by default it should be a translatable string. When I changed language I saw Browse kept untranslated and looked in the Wordpress core translation strings what is available.
It doesn't block anyone using this class to use something else. i.e. I use "Choose Image" as I know I want an image which is also translated.
Maybe it also fits as it opens the media browser anyway.

Maybe should also separate between type=file and type=media/image and have 2 callbacks.

tareq1988 commented:
For file uploads, we should have a configurable label. Right now it's hardcoded and the change you made also hardcoded as well. Instead we should pass the text as a parameter and it'll fallback to `__( 'Choose File' );` if none given.

roidayan commented:
ok. added option to pass button label and the default is Choose File which is translated according to Wordpress language.
Also updated the example with this.

tareq1988 commented on src/class.settings-api.php:
```diff
@@ -343,9 +343,14 @@ function callback_file( $args ) {
         $size = isset( $args['size'] ) && !is_null( $args['size'] ) ? $args['size'] : 'regular';
         $id = $args['section']  . '[' . $args['id'] . ']';
+        if ( isset( $args['options'] ) && isset( $args['options']['button_label'] ) ) {
+            $label = $args['options']['button_label'];
```
Can you please use ternary operator instead of using a big if/else block?

roidayan commented:
did it at first but it was pretty big by itself so didn't think it's needed as it didn't add readability. but ok.

roidayan commented:
its pretty big I think if-else is more readable for this case. do you still want it like this ?

```
$label = ( isset( $args['options'] ) && isset( $args['options']['button_label'] ) ) ?
						$label = $args['options']['button_label'] :
						$label = __( 'Choose File' );
```

tareq1988 commented:
Checking only `isset( $args['options']['button_label'])` should work, no?

roidayan commented:
i'll check if it doesn't give warning in debug

roidayan commented:
only  isset( $args['options']['button_label'])  works good. i'll update the commit. thanks

roidayan commented:
updated

tareq1988 commented on src/class.settings-api.php:
```diff
@@ -343,9 +343,12 @@ function callback_file( $args ) {
         $value = esc_attr( $this->get_option( $args['id'], $args['section'], $args['std'] ) );
         $size = isset( $args['size'] ) && !is_null( $args['size'] ) ? $args['size'] : 'regular';
         $id = $args['section']  . '[' . $args['id'] . ']';
+        $label = isset( $args['options']['button_label'] ) ?
```
It should be like this:
```php
$label = isset( $args['options']['button_label'] ) ? $args['options']['button_label'] : __( 'Choose File' );
```
Don't need to use `$label` on the ternary operator.

roidayan replied:
oh damn. didn't notice it.
